src/viam/sdk/robot/client.hpp

```diff
@@ -23,13 +25,62 @@ namespace viam {
 namespace sdk {
 
 using grpc::Channel;
-using viam::common::v1::ResourceName;
-using viam::common::v1::Transform;
-using viam::robot::v1::FrameSystemConfig;
 using viam::robot::v1::RobotService;
-using viam::robot::v1::Status;
+using time_point = std::chrono::time_point<long long, std::chrono::nanoseconds>;
+
+struct discovery_query {
```

Comment:
Hmm... I personally am not a fan of having a `Robot` base class. In my mind there's only a single canonical implementation of how one should interact with a robot as a client in C++, and so it doesn't really make sense to expose an abstract API the same way it does for the various resources.

That said, there's something very slightly smelly about having these types exist on `RobotClient` (as opposed to `Robot`) that I'm having trouble putting my finger on. So, I'm certainly open to being convinced that an abstract `Robot` base class makes sense.

Either way, I agree that these structs should exist within `RobotClient` (or `Robot` if we create it) rather than as free-floating types.

Reply:
Yes I share your unease, and I find the lack of consistency between `Robot` and the other resource types a little troublesome as well. Also unclear what the use case is for `RobotService_`. I can't quite tell what it is for, which makes it hard for me to honestly evaluate whether a `Robot` base is needed. It feels like if `RobotService_` is meaningful, shouldn't be what it is serving be an implementation of `Robot`, not a `RobotClient`?